app/charts/shared/chart-data-filters.tsx

```diff
@@ -334,17 +336,16 @@ const DataFilterTemporalDimension = ({
   } = dimension;
 
   const formatLocale = useTimeFormatLocale();
-  const timeIntervalWithProps = React.useMemo(
-    () =>
-      getTimeIntervalWithProps(
-        options[0].value as string,
-        options[1].value as string,
-        timeUnit,
-        timeFormat,
-        formatLocale
-      ),
-    [options, timeUnit, timeFormat, formatLocale]
-  );
+  const timeIntervalWithProps = React.useMemo(() => {
+    return getTimeIntervalWithProps(
+      options[0].value as string,
+      options[options.length - 1].value as string,
+      timeUnit,
+      timeFormat,
+      formatLocale
+    );
+  }, [options, timeUnit, timeFormat, formatLocale]);
```

Comment:
It seems you would like the functionality "Add {} to arrow function" from the "JS Refactoring" extension 😁 

<img width="505" alt="image" src="https://github.com/visualize-admin/visualization-tool/assets/465582/c85de33f-aaa1-4dd7-8db1-af9dbde8481a">

<img width="150" alt="image" src="https://github.com/visualize-admin/visualization-tool/assets/465582/1f230308-d553-4db9-9c2d-c3e6a54f2e9a">

Reply:
Ooo, definitely! 👀